### PR TITLE
rename TypeKind::Named to TypeKind::TypeParam

### PR DESCRIPTION
**Pull request by anna-liao**
fixes https://github.com/rust-lang-nursery/rust-bindgen/issues/914. r? @fitzgen 

I made the changes and `cargo build` and `cargo test` run ok.  However, "testing generated bindings" fails.  I checked and it seems to be failing even before my changes?

$ cd tests/expectations
$ cargo test

### CODE REVIEW

**Comment by fitzgen** [comment]
> However, "testing generated bindings" fails. I checked and it seems to be failing even before my changes?

Huh. Our CI is still passing, and I can run them locally just fine, so I suspect this is a build env issue. Do you have the `LIBCLANG_PATH` var set while running these tests as well?

File: src/ir/ty.rs [hunk 1]
```diff
@@ -97,10 +97,10 @@ impl Type {
         }
     }
 
-    /// Is this type of kind `TypeKind::Named`?
+    /// Is this type of kind `TypeKind::TypeParam`?
     pub fn is_named(&self) -> bool {
```
Comment (fitzgen) [inline src/ir/ty.rs]: We should rename `is_named` to `is_type_param` as well.

File: src/ir/ty.rs [hunk 2]
```diff
@@ -259,7 +259,7 @@ impl Type {
     /// tests/headers/381-decltype-alias.hpp
     pub fn is_invalid_named_type(&self) -> bool {
```
Comment (fitzgen) [inline src/ir/ty.rs]: This should become `is_invalid_type_param`.

File: src/ir/ty.rs [hunk 3]
```diff
@@ -464,14 +464,14 @@ impl DotAttributes for TypeKind {
 
 #[test]
 fn is_invalid_named_type_valid() {
```
Comment (fitzgen) [inline src/ir/ty.rs]: These tests should replace `named_type` with `type_param` in their function names.

**Comment by anna-liao** [comment]
thanks for the review!  I will update the pull request soon.

Yup, I ran the tests with `LIBCLANG_PATH` set.  I couldn't find `libclang.so` even searching recursively through that directory, so I didn't set `DYLD_LIBRARY_PATH`.

<details>
  <summary>tests_expectations output</summary>
   Compiling tests_expectations v0.1.0 (file:///Users/annaliao/repos/rust-bindgen/tests/expectations)
error: unions are unstable and possibly buggy (see issue #32836)
  --> tests/layout_mbuf.rs:90:1
   |
90 | / pub union rte_mbuf__bindgen_ty_1 {
91 | |     /// < Atomically accessed refcnt
92 | |     pub refcnt_atomic: rte_atomic16_t,
93 | |     /// < Non-atomically accessed refcnt
94 | |     pub refcnt: u16,
95 | |     _bindgen_union_align: u16,
96 | | }
   | |_^

error: unions are unstable and possibly buggy (see issue #32836)
   --> tests/layout_mbuf.rs:124:1
    |
124 | / pub union rte_mbuf__bindgen_ty_2 {
125 | |     /// < L2/L3/L4 and tunnel information.
126 | |     pub packet_type: u32,
127 | |     pub __bindgen_anon_1: rte_mbuf__bindgen_ty_2__bindgen_ty_1,
128 | |     _bindgen_union_align: u32,
129 | | }
    | |_^

error: unions are unstable and possibly buggy (see issue #32836)
   --> tests/layout_mbuf.rs:458:1
    |
458 | / pub union rte_mbuf__bindgen_ty_3 {
459 | |     /// < RSS hash result if RSS enabled
460 | |     pub rss: u32,
461 | |     /// < Filter identifier if FDIR enabled
...   |
467 | |     _bindgen_union_align: [u32; 2usize],
468 | | }
    | |_^

error: unions are unstable and possibly buggy (see issue #32836)
   --> tests/layout_mbuf.rs:477:1
    |
477 | / pub union rte_mbuf__bindgen_ty_3__bindgen_ty_1__bindgen_ty_1 {
478 | |     pub __bindgen_anon_1: rte_mbuf__bindgen_ty_3__bindgen_ty_1__bindgen_ty_1__bindgen_ty_1,
479 | |     pub lo: u32,
480 | |     _bindgen_union_align: u32,
481 | | }
    | |_^

error: unions are unstable and possibly buggy (see issue #32836)
   --> tests/layout_mbuf.rs:641:1
    |
641 | / pub union rte_mbuf__bindgen_ty_4 {
642 | |     /// < Can be used for external metadata
643 | |     pub userdata: *mut ::std::os::raw::c_void,
644 | |     /// < Allow 8-byte userdata on 32-bit
645 | |     pub udata64: u64,
646 | |     _bindgen_union_align: u64,
647 | | }
    | |_^

error: unions are unstable and possibly buggy (see issue #32836)
   --> tests/layout_mbuf.rs:675:1
    |
675 | / pub union rte_mbuf__bindgen_ty_5 {
676 | |     /// < combined for easy fetch
677 | |     pub tx_offload: u64,
678 | |     pub __bindgen_anon_1: rte_mbuf__bindgen_ty_5__bindgen_ty_1,
679 | |     _bindgen_union_align: u64,
680 | | }
    | |_^

error: aborting due to 6 previous errors

error: Could not compile `tests_expectations`.
Build failed, waiting for other jobs to finish...
error: build failed

</details>

**Comment by anna-liao** [comment]
Thanks for the guidance to fix the remaining "named" occurrences.

I ran "rust fmt" and it flagged a bunch of line length issues, some in files that I didn't modify.

**Comment by fitzgen** [comment]
> I couldn't find libclang.so even searching recursively through that directory

Ok, that's (one of?) the issue(s?). `LIBCLANG_PATH` needs to point to the directory that contains `libclang.so` or else `bindgen` is going to have a bad time.

> error: unions are unstable and possibly buggy (see issue #32836)

Have you run `rustup update stable` in the last few weeks? The latest stable release enabled `union`s.

**Comment by fitzgen** [comment]
> I ran "rust fmt" and it flagged a bunch of line length issues, some in files that I didn't modify.

That's fine, don't worry about it.

**Comment by anna-liao** [comment]
Thanks for the review @fitzgen!  I squashed the commits, but I'm not sure if I handled the merge conflicts correctly.

Regarding the `libclang.so`, I ran `rustup update stable` and still not finding `libclang.so` along the directory tree where I'd expect it.  I'm trying a full search to see if it is somewhere else.

**Comment by fitzgen** [comment]
Thanks @anna-liao ! I reworded the commit message to be a little more descriptive (you're still marked as author ;) ), and it is landing in https://github.com/rust-lang-nursery/rust-bindgen/pull/924

Thanks again!

**Pull request closed without merging**